include/beast/websocket/detail/pmd_extension.hpp

```diff
@@ -0,0 +1,467 @@
+    {
+    case -1:
+        // extension parameter is present with no value
+        config.client_max_window_bits =
+            o.client_max_window_bits;
+        if(config.client_max_window_bits < 15)
+        {
+            s += "client_max_window_bits=";
+            s += std::to_string(
+                config.client_max_window_bits);
+        }
+        break;
+
+    case 0:
+        /*  extension parameter is absent.
+
+            If a received extension negotiation offer doesn't have the
+            "client_max_window_bits" extension parameter, the corresponding
+            extension negotiation response to the offer MUST NOT include the
+            "client_max_window_bits" extension parameter.
+        */
+        if(o.client_max_window_bits == 15)
+            config.client_max_window_bits = 15;
+        else
+            config.accept = false;
+        break;
+
+    default:
+        // extension parameter has value in [8..15]
+        if(o.client_max_window_bits <
+           offer.client_max_window_bits)
+        {
+            // Use server's lower configured limit
+            config.client_max_window_bits =
+                o.client_max_window_bits;
+            s += "client_max_window_bits=";
+            s += std::to_string(
+                config.client_max_window_bits);
+        }
+        else
+        {
+            config.client_max_window_bits =
+                offer.client_max_window_bits;
+        }
+        break;
+    }
+    if(config.accept)
+        fields.replace("Sec-WebSocket-Extensions", s);
+}
+
+// Normalize the server's response
+//
+inline
+void
+pmd_normalize(pmd_offer& offer)
+{
+    if(offer.accept)
+    {
+        if( offer.server_max_window_bits == 0)
+            offer.server_max_window_bits = 15;
+
+        if( offer.client_max_window_bits ==  0 ||
+            offer.client_max_window_bits == -1)
+            offer.client_max_window_bits = 15;
+    }
+}
+
+//--------------------------------------------------------------------
+
+// Decompress into a DynamicBuffer
+//
+template<class InflateStream, class DynamicBuffer>
+void
+inflate(
+    InflateStream& zi,
+    DynamicBuffer& dynabuf,
+    boost::asio::const_buffer const& in,
+    error_code& ec)
+{
+    using boost::asio::buffer_cast;
+    using boost::asio::buffer_size;
+    zlib::z_params zs;
+    zs.avail_in = buffer_size(in);
+    zs.next_in = buffer_cast<void const*>(in);
+    for(;;)
+    {
+        auto const bs = dynabuf.prepare(
+            read_size_helper(dynabuf, 65536));
+        auto const out = *bs.begin();
+        zs.avail_out = buffer_size(out);
+        zs.next_out = buffer_cast<void*>(out);
+        zi.write(zs, zlib::Flush::sync, ec);
+        dynabuf.commit(zs.total_out);
+        zs.total_out = 0;
+        if( ec == zlib::error::need_buffers ||
+            ec == zlib::error::end_of_stream)
+        {
+            ec = {};
+            break;
+        }
+        if(ec)
+            return;
```

Comment:
The use of `break` and `return` do the same thing, but spelling them differently like this doesn't give me confidence that the code is doing what the author intended.  I recommend using `break` in both places or `return` in both places.

Also it might be nice to combine this like so:
````
       if (ec)
       {
             if( ec == zlib::error::need_buffers ||
                 ec == zlib::error::end_of_stream)
                 ec = {};
             return;
       }
````